source/faucet/main.d

```diff
@@ -185,13 +185,30 @@ public void setup (ref State state, API client, uint count)
 
 *******************************************************************************/
 
-void send (API client, ref State state) @safe
+void send (API client, ref State state)
 {
     state.update(client, Height(state.known + 1));
     if (state.known < 1)
         return logInfo("Waiting for setup to be completed");
 
-    logInfo("About to send transactions, UTXO set: %d entries", state.utxos.length);
+    logInfo("About to send transactions...");
+
+    // Sort them so we don't iterate multiple time
+    // Note: This may cause a lot of memory usage, might need restructuing later
+    // Mutable because of https://issues.dlang.org/show_bug.cgi?id=9792
+    auto sutxo = state.utxos.values.sort!((a, b) => a.output.value < b.output.value);
+    const size = sutxo.length();
+    logInfo("\tUTXO set: %d entries", size);
+
+    immutable median = sutxo[size / 2].output.value;
+    // Should be 500M (5,000,000,000,000,000) for the time being
+    immutable sum = sutxo.map!(utxo => utxo.output.value)
+        .fold!((a, b) => Amount(a).mustAdd(b))(Amount(0));
+    assert(sum == Amount.MaxUnitSupply);
+    auto mean = Amount(sum); mean.div(size);
```

Comment:
Interesting syntax. When should such a syntax be used?

Reply:
It's nothing special really. You could write all your code on a single line, whitespace and newline rarely matters to the compiler in C-like languages. I did it here because there is no way to initialize `mean` to the right value (you have to initialize  then modify), hence I wanted to notice "hey, this is really just initialization".

Slightly off-topic: https://en.wikipedia.org/wiki/Minification_(programming) is a must do to reduce the size of webpage.